test/gilded-rose.spec.ts

```diff
@@ -1,9 +1,190 @@
 import { Item, GildedRose } from '@/gilded-rose';
 
 describe('Gilded Rose', () => {
-  it('should foo', () => {
-    const gildedRose = new GildedRose([new Item('foo', 0, 0)]);
-    const items = gildedRose.updateQuality();
-    expect(items[0].name).toBe('fixme');
-  });
+  describe('Generic Items', () => {
+    it('should foo', () => {
+      const gildedRose = new GildedRose([new Item('foo', 0, 0)]);
+      const items = gildedRose.updateQuality();
+      expect(items[0].name).toBe('foo');
+    });
+    it('Given an item, when updateQuality is called, then sellin and quality lowered.', () => {
+      // Given
+      const testEnv = new GildedRose([new Item('item1', 5, 5)]);
+
+      // When
+      const items = testEnv.updateQuality();
+
+      // Then
+      expect(items[0].quality).toBeLessThan(5);
+      expect(items[0].sellIn).toBeLessThan(5);
+    })
+
+    it('Given 2 items, item1 passed sellIn and item2 not passed sellIn, when updateQuality is called, then quality drop for item1 must be twice quality drop for item2.', () => {
+      // Given
+      const testEnv = new GildedRose([new Item('item1', -1, 5), new Item('item2', 4, 5)]);
+
+      // When
+      const items = testEnv.updateQuality();
+
+      // Then
+      expect(5 - items[0].quality).toBe(2 * (5 - items[1].quality));
+    })
+  })
```

Comment:
I'm not sure how I feel about this test case, I think you're trying to test 2 things at once and might be better to just have 2 test cases, testing item 0's quality and item 1's quality explicitly.